kube-controllers/pkg/controllers/node/ipam.go

```diff
@@ -568,16 +574,26 @@ func (c *ipamController) checkEmptyBlocks() error {
 			continue
 		}
 
+		// During a Flannel migration, we can only migrate blocks affined to nodes that have already undergone the migration
+		migrating, err := c.nodeIsBeingMigrated(node)
+		if err != nil {
+			logc.Warn("Couldn't find node affined to empty block, skipping affinity release")
+			continue
+		}
+		if migrating {
+			logc.Info("Node affined to block is currently undergoing a migration from Flannel, skipping affinity release")
+		}
+
 		// Find the actual block object.
-		block, ok := c.allBlocks[blockCIDR]
-		if !ok {
+		block, migrating := c.allBlocks[blockCIDR]
```

Comment:
Is this an unintended change? I think this should go back to being `block, ok := c.allBlocks[blockCIDR]`

Reply:
yes, updated it.